### PR TITLE
BOAC-3238 Standardize representation of appointment status and events

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/BOAC-3238

### CODE REVIEW

File: boac/models/appointment.py [hunk 1]
```diff
@@ -115,7 +115,8 @@ def find_advisors_by_name(cls, tokens, limit=None):
             token_conditions.append(
                 f"""JOIN appointments a{idx}
                 ON UPPER(a{idx}.advisor_name) LIKE :token_{idx}
-                AND a{idx}.advisor_uid = a.advisor_uid""",
+                AND a{idx}.advisor_uid = a.advisor_uid
+                AND a{idx}.advisor_name = a.advisor_name""",
```
Comment (paulkerschen) [inline boac/models/appointment.py]: Fixes an edge case where, if the advisor name associated with a UID changes, the non-matching name would come back in results.

File: tests/util.py [hunk 2]
```diff
@@ -54,13 +54,17 @@ def override_config(app, key, value):
     """Temporarily override an app config value."""
     old_value = app.config[key]
     app.config[key] = value
-    yield
-    app.config[key] = old_value
+    try:
+        yield
+    finally:
+        app.config[key] = old_value
```
Comment (paulkerschen) [inline tests/util.py]: Ensure that the config resets even if a test fails; otherwise confusing later failures ensue.

**Comment by johncrossman** [comment]
Another hard night of refactoring.

image

Thanks @paulkerschen
